### PR TITLE
executor: CheckIndexRangeExec should set the types of index columns

### PR DESCRIPTION
**Pull request by zz-jason**
### What problem does this PR solve? 

The type of index columns was not properly set in `CheckIndexRangeExec`, which causes certain unexpected behavior if these type information is used in the coprocessor.

### What is changed and how it works?

set the type of index columns according to the types in schema.

### Check List 

Tests 

 - Integration test

Code changes

 - Has exported function/method change

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: executor/admin.go [hunk 1]
```diff
@@ -97,6 +97,9 @@ func (e *CheckIndexRangeExec) Open(ctx context.Context) error {
 		ID:   model.ExtraHandleID,
 		Name: model.ExtraHandleName,
 	})
+	for i := range e.cols {
+		e.cols[i].FieldType = *e.schema.Columns[i].RetType
```
Comment (lysu) [inline executor/admin.go]: Question: this assign to `e.cols[i].FieldType` whether have risk to modify the shared variable `infoschema.infoSchemaDB` ?_?

Comment (zz-jason) [reply executor/admin.go]: nice catch, we should only fix the field type of the newly added column.

**Comment by zz-jason** [comment]
@lysu PTAL

**Pull request merged**